tests/integration/test_metrics.py

```diff
@@ -36,8 +36,16 @@ def test_metrics():
     response_text = retrieve_metrics(client)
 
     # counters that are expected to be part of metrics
-    assert "rest_api_calls_total " in response_text
-    assert "llm_calls_total " in response_text
+    expected_counters = (
+        "rest_api_calls_total",
+        "llm_calls_total",
+        "llm_calls_failures_total",
+        "llm_validation_errors_total",
+    )
+
+    # check if all counters are present
+    for expected_counter in expected_counters:
+        assert f"{expected_counter} " in response_text
```

Comment:
why not just `assert expected_counter in response_text`? Is the extra space important?

Reply:
I would like to be precise, but depends, not having strong opinion there.

Reply:
ok